merlin/models/tf/data_augmentation/negative_sampling.py

```diff
@@ -62,12 +57,19 @@ def __init__(
         self.schema = schema.select_by_tag(Tags.ITEM)
         self.seed = seed
         self.run_when_testing = run_when_testing
-        self.return_tuple = return_tuple
 
-    def call(self, inputs: TabularData, targets=None, testing=False, **kwargs) -> Prediction:
+    def call(
+        self, inputs: TabularData, targets=None, training=False, testing=False, **kwargs
+    ) -> Union[Prediction, Tuple]:
         """Extend batch of inputs and targets with negatives."""
+
+        def get_tuple(x, y):
+            if training or testing:
```

Comment:
If you call model.predict the model would be called without either of those flags, could that be an issue?

Reply:
Updated this check to handle `training or testing or y is None` to handle predict here.